soh/soh/Enhancements/mods.cpp

```diff
@@ -953,6 +953,23 @@ void RegisterAltTrapTypes() {
     });
 }
 
+void RegisterRandomizerSheikSpawn() {
+    GameInteractor::Instance->RegisterGameHook<GameInteractor::OnSceneSpawnActors>([]() {
+        if (!gPlayState) return;
+        bool canSheik = OTRGlobals::Instance->gRandomizer->GetRandoSettingValue(RSK_TRIAL_COUNT) != RO_GANONS_TRIALS_SKIP;
```

Comment:
does it make sense to spawn sheik even when the light arrow hint is turned off? i think i'd vote to have sheik only appear when the setting is on.

Reply:
It should be a combination of the two, as Sheik relies on LA hint *and* there being trials (otherwise you go to Ganondorf for LA hint).